File: LeetCode/1684. 统计一致字符串的数目.js
/**
 * @param {string} allowed
 * @param {string[]} words
 * @return {number}
 */
var countConsistentStrings = function (allowed, words) {
  let count = 0
  const n = words.length
  for (let i = 0; i < n; i++) {
    const wordArr = words[i].split('')
    let flag = true
    for (let j = 0; j < wordArr.length; j++) {
      if (!allowed.includes(wordArr[j])) flag = false
    }
    if (flag) count++
  }
  return count
}

File: LeetCode/1684. 统计一致字符串的数目.c
#include <stdio.h>
int countConsistentStrings(char *allowed, char **words, int wordsSize)
{
    int mask = 0;
    for (int i = 0; allowed[i] != '\0'; i++)
    {
        mask |= 1 << (allowed[i] - 'a');
    }
    int count = 0;
    for (int i = 0; i < wordsSize; i++)
    {
        int mask1 = 0;
        for (int j = 0; words[i][j] != '\0'; j++)
        {
            mask1 |= 1 << (words[i][j] - 'a');
        }
        if ((mask1 | mask) == mask)
            count++;
    }
    return count;
}
/*
0000 01
如果是b b -a = 1  左移 0000 01 << 0000 01 0000 01
如果是c c - a  = 2 左移 0000 01 << 0000 10 0000 11
 */

char *a = "ab";
char *b[] = {"ad", "bd", "aaab", "baa", "badab"};
countConsistentStrings(a, b);